lib/shopify_api/resources/checkout.rb

```diff
@@ -1,4 +1,31 @@
 module ShopifyAPI
   class Checkout < Base
+    def id
+      @attributes[:token]
+    end
+
+    def pay(session_params, &block)
+      vault_session = VaultSession.create(session_params)
+      pay_with(vault_session, &block)
+    end
+
+    def pay_with(vault_session, &block)
+      Payment.create_from(self, vault_session).tap(&block)
+    end
+
+    class ShippingRate < Base
```

Comment:
Shipping rate should live a level higher: it's not a `CheckoutShippingRate`/`Checkout::ShippingRate`, but simply `ShippingRate`. In the future other resources are likely to share the same shipping-rate interface.